File: ruby/arrays/hashes/iterating_over_hashes_and_reading_elements.rb
#!/usr/bin/ruby

hash = {
    2 => "Hello, there!",
    'color' => 'orange',
    3.14 => 'PI',
    'season' => 'winter',
    2.72 => 'e',
}

for element in hash
  puts "element: #{hash}"
end

puts()

for key in hash.keys()
  puts "hash[#{key}]: #{hash[key]}"
end

puts()

for value in hash.values()
  puts "value: #{value}"
end

puts()

hash.each do |key, value|
  puts "hash[#{key}]: #{hash[key]} (the same: #{value})"
end

puts()

hash.each do |element|
  puts "element: #{element}"
end

puts()

hash.each_key do |key|
  puts "hash[#{key}]: #{hash[key]}"
end

puts()

File: perl/arrays/hashes/iterating_over_hashes_and_reading_elements.pl
#
# Copyright (C) 2019-2026 Katarzyna Krasińska
# Languagium - https://github.com/katheroine/languagium
# Licensed under GPL-3.0 - see LICENSE.md
#

use strict;
use warnings;

my %hash = (
  2 => "Hello, there!",
  'color' => 'orange',
  3.14 => 'PI',
  'season' => 'winter',
  2.72 => 'e',
);

# for can be used instead of foreach

foreach my $key (keys %hash) {
  print "\$hash{$key}: $hash{$key}\n";
}

print("\n");

while ((my $key, my $value) = each (%hash)) {
  print "\$hash{$key}: $hash{$key} (the same: $value)\n";
}

print("\n");
